deploy-caplin-erigon.py

```
@@ -0,0 +1,839 @@
+
+        # Extract the binary to the home folder
+        with tarfile.open(tar_filename, "r:gz") as tar:
+            def is_within_directory(directory, target):
+                abs_directory=os.path.abspath(directory)
+                abs_target=os.path.abspath(target)
+                return os.path.commonpath([abs_directory]) == os.path.commonpath([abs_directory, abs_target])
+
+            for member in tar.getmembers():
+                member_path = os.path.join(os.getcwd(), member.name)
+                if not is_within_directory(os.getcwd(), member_path):
+                    raise Exception("Attempted Path Traversal in Tar File")
```

Ruff (0.14.4)

Create your own exception

(TRY002)